### PR TITLE
Found withdraw internal contract creation

### PR DESCRIPTION
**Pull request by Griffics**

### CODE REVIEW

File: Context/InternalBlockchain/Repository/RepositoryInterface.ts [hunk 1]
```diff
@@ -16,6 +18,13 @@ export default interface RepositoryInterface {
...
     disconnect: () => void;
     burnAsset: (amount: string) => void;
-    getAsset: () => Promise<any>;
     getBurnOperations: (account: string) => Promise<OperationBurn[]>;
+    getInternalAsset: () => Promise<Map<string, any>>;
+    getAsset: (assetId: string) => Promise<Map<string, any>>;
+    getAccountHistory: () => Promise<WithdrawTransaction[]>;
+    getAccount(accountId: string): Promise<Map<string, any>>;
+    getEesAccount: () => Promise<Map<string, any>>;
+    getBurnOperations: (account: string) => Promise<OperationBurn[]>;
```
Comment (walteratwork) [inline Context/InternalBlockchain/Repository/RepositoryInterface.ts]: duplicate

File: Context/config.ts [hunk 2]
```diff
@@ -18,6 +18,7 @@ const config = {
         network: process.env.ETH_NETWORK_NAME as string,
         private_key: process.env.ETH_PRIVATE_KEY as string,
         minimum_deposit_amount: Web3.utils.toBN(Web3.utils.toWei(process.env.MINIMUM_DEPOSIT_AMOUNT as string)),
+        minimum_withdraw_amount: parseFloat(process.env.MINIMUM_DEPOSIT_AMOUNT as string),
```
Comment (walteratwork) [inline Context/config.ts]: MINIMUM_WITHDRAW_AMOUNT

File: Context/Application/Command/InternalBlockchain/CheckInternalWithdrawalOperation/CheckInternalWithdrawalOperationHandler.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        const normalizedAmount = this.normalizer.normalize(
+            htlcOperation.get("op")[1].amount.amount,
+            await this.internalBlockchain.getAsset(htlcOperation.get("op")[1].amount.asset_id)
+        );
+
+        if (normalizedAmount < config.eth.minimum_withdraw_amount) {
+            throw new Errors.InvalidAmount(withdraw);
+        }
+
+        if (htlcOperation.get("op")[1].amount.asset_id != "1.3.1") {
```
Comment (walteratwork) [inline Context/Application/Command/InternalBlockchain/CheckInternalWithdrawalOperation/CheckInternalWithdrawalOperationHandler.ts]: get ID from blockchain

File: Context/Application/Command/InternalBlockchain/CheckInternalWithdrawalOperation/CheckInternalWithdrawalOperationHandler.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+            throw new Errors.InvalidPreimage(withdraw);
+        }
+    }
+
+    private async checkTransferOperation(transferOperation: Map<string, any>, withdraw: Withdraw) {
+        await this.checkLastIrreversible(transferOperation);
+        await this.checkReceiver(transferOperation);
+
+        let minimalWithdrawalFee = config.revpop.rveth_withdrawal_fee;
+        if (transferOperation.get("op")[1].amount.asset_id == "1.3.0") {
```
Comment (walteratwork) [inline Context/Application/Command/InternalBlockchain/CheckInternalWithdrawalOperation/CheckInternalWithdrawalOperationHandler.ts]: get ID from blockchain

File: Context/Application/Command/InternalBlockchain/CheckInternalWithdrawalOperation/CheckInternalWithdrawalOperationHandler.ts [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        }
+    }
+
+    private async checkLastIrreversible(operation: Map<string, any>) {
+        if (!this.lastIrreversibleBlockNumber) {
+            this.lastIrreversibleBlockNumber = await this.internalBlockchain.getLastIrreversibleBlockNumber();
+        }
+
+        if (operation.get("block_num") > this.lastIrreversibleBlockNumber) {
+            throw new Errors.BlockIsReversible(operation.get("id"));
```
Comment (walteratwork) [inline Context/Application/Command/InternalBlockchain/CheckInternalWithdrawalOperation/CheckInternalWithdrawalOperationHandler.ts:111]: Do not hard fail here. Need to check the contract later.

**Review by walteratwork — CHANGES_REQUESTED** [review]
Pls rebase: many commits in the branch are already merged in development.

File: Context/Application/Command/InternalBlockchain/CheckInternalWithdrawalOperation/CheckInternalWithdrawalOperationHandler.ts (hunk 5 above)
Comment (Griffics) [reply Context/Application/Command/InternalBlockchain/CheckInternalWithdrawalOperation/CheckInternalWithdrawalOperationHandler.ts:111]: It is not hard fail, contracts will rechecked on next run

**Review by walteratwork — APPROVED** [review]

**Pull request merged**